### PR TITLE
docs: Add instructions for running LVH against custom kernel

### PR DESCRIPTION
**Pull request by brb**

### CODE REVIEW

File: Documentation/contributing/testing/e2e.rst [hunk 1]
```diff
@@ -166,3 +165,45 @@ To stop the VM, run from the host:
...
+required once `little-vm-helper#117 <https://github.com/cilium/little-vm-helper/issues/117>`_
+has been resolved):
+
+.. code-block:: shell-session
+
+    $ ssh -p 2222 -o "StrictHostKeyChecking=no" root@localhost
+    # cd /host/bpf-next
+    # make modules_install
+
+Finally, you can use instructions from the previous chapter to run and to test Cilium.
```
Comment (ghost) [inline Documentation/contributing/testing/e2e.rst]: Could you link the relevant section instead of saying "the previous chapter"?

**Review by ghost — CHANGES_REQUESTED** [review]
A localization nit, otherwise looks good.

File: Documentation/contributing/testing/e2e.rst (hunk 1 above)
Comment (brb) [reply Documentation/contributing/testing/e2e.rst]: Aye, good idea! Let me figure out how to link a chapter in RST :sweat_smile:

File: Documentation/contributing/testing/e2e.rst [hunk 2]
```diff
@@ -166,3 +167,45 @@ To stop the VM, run from the host:
...
+
+It is possible to test Cilium on a LVH VM with a custom built Linux kernel (e.g.,
+for fast testing iterations when doing kernel development work for Cilium features).
+
+First, to configure and to build the kernel:
+
+.. code-block:: shell-session
+
+   $ git clone --depth=1 https://git.kernel.org/pub/scm/linux/kernel/git/bpf/bpf-next.git
+   $ cd bpf-next/ # hack hack hack
```
Comment (bimmlerd) [inline Documentation/contributing/testing/e2e.rst]: `hack hack hack` as in this is a hack or do some hack hack hacking on it :grin:

File: Documentation/contributing/testing/e2e.rst [hunk 3]
```diff
@@ -166,3 +167,45 @@ To stop the VM, run from the host:
 .. code-block:: shell-session
 
     $ pkill qemu-system-x86
+
+Running tests in VM with custom kernel
```
Comment (bimmlerd) [inline Documentation/contributing/testing/e2e.rst]:
```suggestion
Running tests in a VM with a custom kernel
```

File: Documentation/contributing/testing/e2e.rst [hunk 4]
```diff
@@ -166,3 +167,45 @@ To stop the VM, run from the host:
...
+""""""""""""""""""""""""""""""""""""""
+
+It is possible to test Cilium on a LVH VM with a custom built Linux kernel (e.g.,
```
Comment (bimmlerd) [inline Documentation/contributing/testing/e2e.rst]:
```suggestion
It is possible to test Cilium on an LVH VM with a custom built Linux kernel (for example,
```

File: Documentation/contributing/testing/e2e.rst [hunk 5]
```diff
@@ -166,3 +167,45 @@ To stop the VM, run from the host:
...
+Second, start the LVH VM with the custom kernel:
+
+.. code-block:: shell-session
+
+   $ lvh run --image ./images/kind_bpf-next.qcow2 \
+        --host-mount $(pwd) \
+        --kernel ./bpf-next/arch/x86_64/boot/bzImage \
+        --daemonize -p 2222:22 --cpu=3 --mem=6G \
+
+Third, SSH into VM, and install the custom kernel modules (the step is no longer
```
Comment (bimmlerd) [inline Documentation/contributing/testing/e2e.rst]:
```suggestion
Third, SSH into the VM, and install the custom kernel modules (this step is no longer
```

File: Documentation/contributing/testing/e2e.rst [hunk 6]
```diff
@@ -166,3 +167,45 @@ To stop the VM, run from the host:
...
+required once `little-vm-helper#117 <https://github.com/cilium/little-vm-helper/issues/117>`_
+has been resolved):
+
+.. code-block:: shell-session
+
+    $ ssh -p 2222 -o "StrictHostKeyChecking=no" root@localhost
+    # cd /host/bpf-next
+    # make modules_install
+
+Finally, you can use instructions from :ref:`the previous chapter<test_cilium_on_lvh>` to run and to test Cilium.
```
Comment (bimmlerd) [inline Documentation/contributing/testing/e2e.rst]:
```suggestion
Finally, you can use the instructions from :ref:`the previous chapter<test_cilium_on_lvh>` to run and test Cilium.
```

**Review by bimmlerd — CHANGES_REQUESTED** [review]
couple of spelling/grammar fixes

**Review by bimmlerd — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**